contracts/governance/staking/StakedToken.sol

```diff
@@ -301,15 +306,24 @@ contract StakedToken is GamifiedVotingToken, InitializableReentrancyGuard {
 
             // 5. Settle the withdrawal by burning the voting tokens
             _burnRaw(_msgSender(), totalWithdraw, exitCooldown, false);
-            //      Log any redemption fee to the rewards contract
+            // Log any redemption fee to the rewards contract if MTA or
+            // the staking token if mBPT.
             _notifyAdditionalReward(totalWithdraw - userWithdrawal);
-            //      Finally transfer tokens back to recipient
-            STAKED_TOKEN.safeTransfer(_recipient, userWithdrawal);
+            // Finally transfer staked tokens back to recipient
+            _withdrawStakedTokens(_recipient, userWithdrawal);
 
             emit Withdraw(_msgSender(), _recipient, _amount);
         }
     }
 
+    /**
+     * @dev Transfers an `amount` of staked tokens to the withdraw `recipient`. eg MTA or mBPT.
+     * Can be overridden if the tokens are held elsewhere. eg in the Balancer Pool Gauge.
+     */
+    function _withdrawStakedTokens(address _recipient, uint256 amount) internal virtual {
+        STAKED_TOKEN.safeTransfer(_recipient, amount);
```

Comment:
Should the ` emit Withdraw(_msgSender(), _recipient, _amount);` be included in this method ?
Second thought , if the method is override and the event is not emitted then the behaviour would not be constant.

Reply:
It could be moved into `_withdrawStakedTokens` but there's a risk the implementation will not emit the event so I left the emits in `_withdraw`